src/main/java/com/notably/logic/commands/suggestion/SearchSuggestionCommand.java

```diff
@@ -0,0 +1,94 @@
+    }
+
+    @Override
+    public void execute(Model model) {
+        // Nullity check
+        Objects.requireNonNull(model);
+
+        // Set response text
+        model.setResponseText(RESPONSE_MESSAGE);
+
+        // Set suggestions
+        List<SuggestionItem> suggestions = traverseTree(model);
+
+        model.setSuggestions(suggestions);
+    }
+
+    /**
+     * Traverses the whole block tree to get the blocks which contains the keyword.
+     * @param model The app's model.
+     * @return List of SuggestionItem with a display text of the block's path and action of opening the block.
+     */
+    private List<SuggestionItem> traverseTree(Model model) {
+        Queue<AbsolutePath> pathQueue = new LinkedList<>();
+        pathQueue.offer(AbsolutePath.fromString("/"));
+
+        List<SuggestionItem> suggestions = new ArrayList<>();
+
+        while (!pathQueue.isEmpty()) {
+            AbsolutePath currentPath = pathQueue.poll();
+            BlockTree blockTree = model.getBlockTree();
+
+            List<BlockTreeItem> childrenBlocks = blockTree.get(currentPath).getBlockChildren();
+            List<AbsolutePath> childrenPaths = childrenBlocks
+                    .stream()
+                    .map(item -> {
+                        List<String> combinedComponents = new ArrayList<>(currentPath.getComponents());
+                        combinedComponents.add(item.getTitle().getText());
+                        AbsolutePath absolutePath = AbsolutePath.fromComponents(combinedComponents);
+
+                        BlockTreeItem blockTreeItem = blockTree.get(absolutePath);
+                        String blockBody = blockTreeItem.getBody().getText();
+
+                        /* If a blockBody contains the keyword, we create SuggestionItem by
+                           counting the keyword's number of occurrences in the blockBody, setting display text,
+                           and setting action to open that particular block when the user chooses that suggestion. */
+                        String bodyLowerCase = blockBody.toLowerCase();
+                        if (bodyLowerCase.contains(keyword.toLowerCase())) {
+                            String[] blockBodies = bodyLowerCase.split(keyword, -1);
+                            int frequency = blockBodies.length - 1;
+                            String displayText = absolutePath.getStringRepresentation();
+                            Runnable action = () -> {
+                                OpenCommand openCommand = new OpenCommand(absolutePath);
+                                openCommand.execute(model);
+                            };
+                            SuggestionItem suggestionItem = new SuggestionItemImpl(displayText, frequency, action);
+                            suggestions.add(suggestionItem);
+                        }
+                        return absolutePath;
+                    })
+                    .collect(Collectors.toList());
```

Comment:
Hmm, I don't think this is the best way of writing this section. However, let's just work with this for now.